cdc/sink/mq_flush_worker.go

```diff
@@ -177,7 +187,17 @@ func (w *flushWorker) asyncSend(
 
 // run starts a loop that keeps collecting, sorting and sending messages
 // until it encounters an error or is interrupted.
-func (w *flushWorker) run(ctx context.Context) error {
+func (w *flushWorker) run(ctx context.Context) (retErr error) {
+	defer func() {
+		select {
+		case w.errCh <- retErr:
+		default:
+			log.Warn("flushWorker error is dropped due to a previous error", zap.Error(retErr))
+		}
+		close(w.errCh)
```

Comment:
Have a question here, could the `default` branch run?
Supposing the err channel is full in L193, the channel could have been closed, the panic `send on closed channel` could happen.

Reply:
The channel is written to exactly once.